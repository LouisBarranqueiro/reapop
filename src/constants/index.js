export const DEFAULT_STATUS = 'default';
export const INFO_STATUS = 'info';
export const SUCCESS_STATUS = 'success';
export const WARNING_STATUS = 'warning';
export const ERROR_STATUS = 'error';

export const STATUS = {
  default: DEFAULT_STATUS,
  info: INFO_STATUS,
  success: SUCCESS_STATUS,
  warning: WARNING_STATUS,
  error: ERROR_STATUS
};

export const TOP = 't';
export const TOP_CENTER = 'tc';
export const TOP_LEFT_POSITION = 'tl';
export const TOP_RIGHT_POSITION = 'tr';
export const BOTTOM = 'b';
export const BOTTOM_CENTER = 'bc';
export const BOTTOM_LEFT_POSITION = 'bl';
export const BOTTOM_RIGHT_POSITION = 'br';

export const POSITIONS = {
  top: TOP,
  topCenter: TOP_CENTER,
  topLeft: TOP_LEFT_POSITION,
  topRight: TOP_RIGHT_POSITION,
  bottom: BOTTOM,
  bottomCenter: BOTTOM_CENTER,
  bottomLeft: BOTTOM_LEFT_POSITION,
  bottomRight: BOTTOM_RIGHT_POSITION
};

// default value for notifications
export const DEFAULT_NOTIFICATION = {
  status: DEFAULT_STATUS,
  position: TOP_RIGHT_POSITION,
  dismissible: true,
  dismissAfter: 5000,
  allowHTML: false,
  closeButton: false
};
