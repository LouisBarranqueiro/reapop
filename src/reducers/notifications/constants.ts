export enum NotificationAction {
    UpsertNotification = 'reapop/upsertNotification',
    DismissNotification = 'reapop/dismissNotification',
    DismissNotifications = 'reapop/dismissNotifications',
}
