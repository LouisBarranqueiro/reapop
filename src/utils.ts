type TimeoutId = ReturnType<typeof setTimeout>

export class Timer {
    private timerId: TimeoutId | undefined
    private start: Date | undefined
    private callback: (...args: any[]) => any
    remainingTime: number

    constructor(delay: number, callback: (...args: any[]) => any) {
        this.remainingTime = delay
        this.callback = callback
    }

    pause() {
        clearTimeout(<TimeoutId>this.timerId)
        this.remainingTime -= new Date().getTime() - (<Date>this.start).getTime()
    }

    resume() {
        this.start = new Date()
        clearTimeout(<TimeoutId>this.timerId)
        this.timerId = setTimeout(this.callback, this.remainingTime)
    }
}

export const clone = <T extends unknown>(origObject: T): T => {
    if (typeof origObject !== 'object' || origObject === null) {
        return origObject
    }

    if (Array.isArray(origObject)) {
        return origObject.map((value) => clone(value)) as T
    }

    const newObject: Record<string, unknown> = {}
    for (const key in origObject) {
        newObject[key] = clone((origObject as Record<string, unknown>)[key])
    }

    return newObject as T
}

export const isObject = (value: unknown) => !!value && (value as any).constructor === Object
