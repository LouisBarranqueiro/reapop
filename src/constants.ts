import {NewNotification, Notification, Position} from './reducers/notifications/types'

export type NotificationConfig = {
    defaultProps: Partial<Notification>
    customizeNotification: ((notification: NewNotification) => void) | null
    generateId: ((notification: NewNotification) => string) | null
}

export const CONFIG: NotificationConfig = {
    defaultProps: {},
    customizeNotification: null,
    generateId: null,
}

export const STATUSES: {
    none: 'none'
    info: 'info'
    success: 'success'
    loading: 'loading'
    warning: 'warning'
    error: 'error'
} = Object.freeze({
    none: 'none',
    info: 'info',
    success: 'success',
    loading: 'loading',
    warning: 'warning',
    error: 'error',
})

export const POSITIONS: {
    topCenter: 'top-center'
    topLeft: 'top-left'
    topRight: 'top-right'
    bottomCenter: 'bottom-center'
    bottomLeft: 'bottom-left'
    bottomRight: 'bottom-right'
} = Object.freeze({
    topCenter: 'top-center',
    topLeft: 'top-left',
    topRight: 'top-right',
    bottomCenter: 'bottom-center',
    bottomLeft: 'bottom-left',
    bottomRight: 'bottom-right',
})

const classnamePrefix = 'reapop__'

export const classnames = {
    container: (position: Position, singleContainer: boolean) => {
        const classes = [`${classnamePrefix}container`, `${classnamePrefix}container--${position}`]
        if (singleContainer) {
            classes.push(`${classnamePrefix}container--single`)
        }
        return classes
    },
    notification: (notification: Notification) => [
        `${classnamePrefix}notification`,
        `${classnamePrefix}notification--${notification.status}`,
        `${classnamePrefix}notification--buttons-${notification.buttons.length}`,
        notification.dismissible ? `${classnamePrefix}notification--dismissible` : '',
    ],
    notificationIcon: `${classnamePrefix}notification-icon`,
    notificationImageContainer: `${classnamePrefix}notification-image-container`,
    notificationImage: `${classnamePrefix}notification-image`,
    notificationDismissIcon: `${classnamePrefix}notification-dismiss-icon`,
    notificationMeta: `${classnamePrefix}notification-meta`,
    notificationTitle: `${classnamePrefix}notification-title`,
    notificationMessage: `${classnamePrefix}notification-message`,
    notificationButtons: `${classnamePrefix}notification-buttons`,
    notificationButton: `${classnamePrefix}notification-button`,
    notificationButtonText: `${classnamePrefix}notification-button-text`,
}
