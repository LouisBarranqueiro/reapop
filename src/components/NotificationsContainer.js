import React, {Component} from 'react';
import TransitionGroup from 'react/lib/ReactCSSTransitionGroup';
import Notification from './Notification';
import {POSITIONS} from '../constants';

export class NotificationsContainer extends Component {
  // default types
  static defaultProps = {
    notifications: []
  };

  // Properties types
  static propTypes = {
    notifications: React.PropTypes.array.isRequired,
    position: React.PropTypes.string.isRequired,
    theme: React.PropTypes.shape({
      notificationsContainer: React.PropTypes.shape({
        className: React.PropTypes.shape({
          main: React.PropTypes.string.isRequired,
          position: React.PropTypes.func.isRequired
        }).isRequired,
        transition: React.PropTypes.shape({
          name: React.PropTypes.object.isRequired,
          enterTimeout: React.PropTypes.number.isRequired,
          leaveTimeout: React.PropTypes.number.isRequired
        }).isRequired
      }).isRequired,
      notification: React.PropTypes.shape({
        className: React.PropTypes.object.isRequired
      }).isRequired
    }).isRequired
  };

  /**
   * Constructor
   * Bind methods
   * @param {Object} props
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this._renderNotifications = this._renderNotifications.bind(this);
  }

  /**
   * Render notifications
   * @private
   * @returns {XML}
   */
  _renderNotifications() {
    // get all notifications and default values for notifications
    const {
      position,
      theme: {
        notification: {className}
      }
    } = this.props;
    let {notifications} = this.props;

    // when notifications are displayed at the bottom,
    // we display notifications from bottom to top
    if ([POSITIONS.bottomLeft, POSITIONS.bottomRight, POSITIONS.bottomCenter, POSITIONS.bottom]
        .indexOf(position) >= 0) {
      notifications = notifications.reverse();
    }

    return notifications.map((notification) => (
      <Notification key={notification.id} notification={notification} className={className}/>
    ));
  }

  /**
   * Render
   * @returns {XML}
   */
  render() {
    const {
      className, transition: {name, enterTimeout, leaveTimeout}
    } = this.props.theme.notificationsContainer;
    const {position} = this.props;

    return (
      <div className={`${className.main} ${className.position(position)}`}>
        <TransitionGroup
          transitionName={name}
          transitionEnterTimeout={enterTimeout}
          transitionLeaveTimeout={leaveTimeout}>
          {this._renderNotifications()}
        </TransitionGroup>
      </div>
    );
  }
}

export default NotificationsContainer;
