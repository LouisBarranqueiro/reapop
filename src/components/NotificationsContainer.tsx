import React from 'react'
import {TransitionGroup} from 'react-transition-group'
import {Notification as NotificationType, Position} from '../reducers/notifications/types'
import {classnames, POSITIONS} from '../constants'
import {useComponentsContext} from '../hooks/useComponentsContext'
import {useTheme} from '../hooks/useTheme'
import SlideTransition from './SlideTransition'
import NotificationContainer from './NotificationContainer'
import NotificationComponent from './Notification'
import {DismissNotification} from './NotificationsSystem'

type Props = {
    notifications: NotificationType[]
    position: Position
    dismissNotification: DismissNotification
    singleContainer: boolean
}

const NotificationsContainer = (props: Props) => {
    const {position, dismissNotification, singleContainer} = props
    const components = useComponentsContext()
    const theme = useTheme()
    const styles = theme ? theme.container(position, singleContainer) : {}
    const classname = classnames.container(position, singleContainer)
    const Transition = components.Transition || SlideTransition
    const Notification = components.Notification || NotificationComponent
    let {notifications} = props

    // when notifications are displayed at the bottom,
    // we display notifications from bottom to top
    if (([POSITIONS.bottomRight, POSITIONS.bottomCenter, POSITIONS.bottomLeft] as string[]).includes(position)) {
        notifications = notifications.slice().reverse()
    }

    return (
        <div style={styles} className={classname.join(' ')}>
            <TransitionGroup>
                {notifications.map((notification) => (
                    <Transition key={notification.id} notification={notification}>
                        <NotificationContainer
                            key={notification.id}
                            notification={notification}
                            dismissNotification={dismissNotification}
                        >
                            <Notification
                                notification={notification}
                                dismissNotification={dismissNotification}
                                theme={theme}
                                components={components}
                            />
                        </NotificationContainer>
                    </Transition>
                ))}
            </TransitionGroup>
        </div>
    )
}

export default NotificationsContainer
