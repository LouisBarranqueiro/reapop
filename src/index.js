import NotificationsSystem from './components/NotificationsSystem';
import {STATUS, POSITIONS, DEFAULT_NOTIFICATION} from './constants';
import reducer, {
  actions,
  types,
  addNotification,
  updateNotification,
  removeNotification,
  removeNotifications
} from './store/notifications';

export {
  // constants
  // default value for notifications
  DEFAULT_NOTIFICATION,
  // all available status
  STATUS,
  // all available positions
  POSITIONS,
  // all action creators
  actions,
  // actions types
  types,
  // reducer
  reducer,
  // action creators separately
  addNotification,
  updateNotification,
  removeNotification,
  removeNotifications
};
// NotificationsSystem React component
export default NotificationsSystem;
