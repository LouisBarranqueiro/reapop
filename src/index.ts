import NotificationsSystem from './components/NotificationsSystem'
import {NotificationsProvider} from './components/NotificationsProvider'
import {STATUSES, POSITIONS} from './constants'
import {notify, dismissNotifications, dismissNotification} from './reducers/notifications/actions'
import FadeTransition from './components/FadeTransition'
import SlideTransition from './components/SlideTransition'
import GrowTransition from './components/GrowTransition'
import reducer from './reducers/notifications/reducer'
import {useNotifications} from './hooks/useNotifications'
import baseTheme from './themes/base'
import atalhoTheme from './themes/atalho'
import wyboTheme from './themes/wybo'
import bootstrapTheme from './themes/bootstrap'

import {Theme} from './themes/types'
import {setUpNotifications} from './services/notifications'

export {
    // themes
    baseTheme,
    atalhoTheme,
    wyboTheme,
    bootstrapTheme,
    // utils
    setUpNotifications,
    useNotifications,
    // constants
    STATUSES,
    POSITIONS,
    //reducers
    reducer,
    // actions
    notify,
    dismissNotifications,
    dismissNotification,
    // components
    FadeTransition,
    SlideTransition,
    GrowTransition,
    NotificationsProvider,
}

// types
export * from './reducers/notifications/types'
export type {Theme}

export default NotificationsSystem
