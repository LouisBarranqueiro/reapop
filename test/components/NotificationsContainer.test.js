import React from 'react';
import {Provider} from 'react-redux';
import {mount, shallow} from 'enzyme';
import {ExpectedNotificationsContainer} from '../utils/expectedComponents';
import {mockStore, genNotifications, checkPropTypes} from '../utils/fixtures';
import NotificationsContainer from '../../src/components/NotificationsContainer';
import {POSITIONS} from '../../src/constants';
import theme from 'reapop-theme-wybo';

describe('<NotificationsContainer/>', () => {
  let notifications = [];
  let store = null;
  const otherProps = {
    theme
  };

  beforeEach('generate a new notification and init store', () => {
    notifications = genNotifications(3);
    store = mockStore({notifications: []});
  });

  it('should validate props', () => {
    const errors = checkPropTypes({
      notifications: [],
      position: POSITIONS.topLeft,
      theme
    }, NotificationsContainer.propTypes);

    expect(errors.notifications).toBeUndefined();
    expect(errors.position).toBeUndefined();
    expect(errors.theme).toBeUndefined();
  });

  it('should not validate props', () => {
    const errors = checkPropTypes({}, NotificationsContainer.propTypes);
    expect(errors.notifications).toBeDefined();
    expect(errors.position).toBeDefined();
    expect(errors.theme).toBeDefined();
  });

  it('should mount with default props', () => {
    const wrapper = mount(
      <NotificationsContainer position={POSITIONS.topLeft} {...otherProps}/>
    );
    const props = wrapper.props();
    expect(props.notifications).toEqual([]);
  });

  it('should render no notifications', () => {
    const wrapper = shallow(
      <NotificationsContainer position={POSITIONS.topLeft} notifications={[]} {...otherProps}/>
    );
    const expectedWrapper = shallow(
      <ExpectedNotificationsContainer position={POSITIONS.topLeft} notifications={[]}
        {...otherProps}/>
    );
    expect(wrapper.html()).toEqual(expectedWrapper.html());
    // `equals()` doesn't work in this case, I don't know why so
    // we use `debug()` to check that `TransitionGroup` tag are present
    // because after being rendered, it becomes a `span` tag
    expect(wrapper.debug()).toEqual(expectedWrapper.debug());
  });

  it('should render notifications', () => {
    const wrapper = mount(
      <Provider store={store}>
        <NotificationsContainer position={POSITIONS.topLeft} notifications={notifications}
          {...otherProps}/>
      </Provider>
    );
    const expectedWrapper = mount(
      <Provider store={store}>
        <ExpectedNotificationsContainer position={POSITIONS.topLeft} notifications={notifications}
          {...otherProps}/>
      </Provider>
    );
    expect(wrapper.html()).toEqual(expectedWrapper.html());
  });

  it('should reverse and render notifications (bottom notifications)', () => {
    // clone array of notifications
    const notifications1 = [...notifications];
    const notifications2 = [...notifications];
    const expectedNotifications = [...notifications].reverse();
    const wrapper = mount(
      <Provider store={store}>
        <NotificationsContainer position={POSITIONS.bottomLeft} notifications={notifications1}
          {...otherProps}/>
      </Provider>
    );
    const expectedWrapper = mount(
      <Provider store={store}>
        <ExpectedNotificationsContainer position={POSITIONS.bottomLeft}
          notifications={notifications2}
          {...otherProps}/>
      </Provider>
    );
    const props = wrapper.find(NotificationsContainer).props();
    expect(props.notifications).toEqual(expectedNotifications);
    expect(wrapper.html()).toEqual(expectedWrapper.html());
  });
});
