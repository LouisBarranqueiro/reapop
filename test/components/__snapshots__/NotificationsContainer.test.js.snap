// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsContainer/> should render no notifications 1`] = `
"<div className=\\"notifications-container notifications-container--tl\\">
  <TransitionGroup component=\\"div\\" childFactory={[Function: childFactory]} />
</div>"
`;

exports[`<NotificationsContainer/> should render notifications 1`] = `
"<div className=\\"notifications-container notifications-container--tl\\">
  <TransitionGroup component=\\"div\\" childFactory={[Function: childFactory]}>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 0, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 1, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 2, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
  </TransitionGroup>
</div>"
`;

exports[`<NotificationsContainer/> should reverse and render notifications (bottom notifications) 1`] = `
"<div className=\\"notifications-container notifications-container--bl\\">
  <TransitionGroup component=\\"div\\" childFactory={[Function: childFactory]}>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 2, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 1, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
    <CSSTransition classNames={{ enter: 'notification-wrapper-enter', exit: 'notification-wrapper-leave' }} timeout={{ enter: 500, exit: 900 }}>
      <Connect(Notification) notification={{ id: 0, title: 'a title', message: 'a message', image: 'https://httpbin.org/image/jpeg', position: 't', status: 'default', dismissible: true, dismissAfter: 0, allowHTML: true, closeButton: true, buttons: [ { name: 'Yes', primary: true, onClick: [Function: mockConstructor] }, { name: 'No', primary: false, onClick: [Function: mockConstructor] } ], onAdd: [Function: mockConstructor], onRemove: [Function: mockConstructor] }} className={{ main: 'notification', wrapper: 'notification-wrapper', meta: 'notification-meta', title: 'notification-title', message: 'notification-message', icon: 'fa notification-icon', imageContainer: 'notification-image-container', image: 'notification-image', status: [Function: status], dismissible: 'notification--dismissible', buttons: [Function: buttons], closeButtonContainer: 'notification-close-button-container', closeButton: 'fa notification-close-button', button: 'notification-button', buttonText: 'notification-button-text' }} />
    </CSSTransition>
  </TransitionGroup>
</div>"
`;
